### ISSUE TITLE
Error: Command /usr/bin/codesign failed with exit code 1

### ISSUE THREAD

**Issue by DongwonU** [openframeworks/openFrameworks#6705]
Hi there,

I got the ‘code sign error’ when I compile my code. What strange is that it will be compiled successfully at the beginning whenever I generate a new project, but after I compile several times this error appeared. I even tried to restart my mac, but this error still exists. I can make sure I set ‘–deep’ in the signing flags. Does anyone know how to fix this?

I am using openframeworks 0.11.1 with macOS Catalina version 10.15.7 and Xcode 11.6.

Following is the detail of the error:
CodeSign bin/particleSysDebug.app
cd /Users/dongwen/Documents/of_v0.11.1_osx_release/apps/myApps/particleSys
export CODESIGN_ALLOCATE=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/codesign_allocate

Signing Identity: “-”

/usr/bin/codesign --force --sign - --deep --entitlements /Users/dongwen/Library/Developer/Xcode/DerivedData/particleSys-hbifleiksqyfsaguziudghmignjm/Build/Intermediates.noindex/particleSys.build/Debug/particleSys.build/particleSysDebug.app.xcent --timestamp=none /Users/dongwen/Documents/of_v0.11.1_osx_release/apps/myApps/particleSys/bin/particleSysDebug.app

/Users/dongwen/Documents/of_v0.11.1_osx_release/apps/myApps/particleSys/bin/particleSysDebug.app: resource fork, Finder information, or similar detritus not allowed
Command /usr/bin/codesign failed with exit code 1

Thanks!

**Comment by ofTheo** [comment]
Hi - I think I have seen this issue before. 

There is some info here: https://stackoverflow.com/questions/39652867/code-sign-error-in-macos-high-sierra-xcode-resource-fork-finder-information

But I think the solution for someone else was this:
```
Temporary solution:
Disable iCloud Drive for the folder where your project is.
```

Let me know if that fixes it for you or if not we can reopen the issue. 
Cheers!

**Comment by ofTheo** [comment]
We could also maybe add the attribute stripping as a build phase to the Xcode project.  
Let me know if this suggestion also works for you: https://stackoverflow.com/a/43094907 

Maybe that would be good to have in the template.

**Comment by DongwonU** [comment]
Hi Theo,
Thank you for your quick response. I have tried all the methods you mentioned above. Following is the process:

**1. In terminal, direct to the root of the project and run 'xattr -cr .'**
It works, but after several times of compiling, the error pop up again.

**2. I tried the method here [https://stackoverflow.com/a/43094907](https://stackoverflow.com/a/43094907).**
As I don't have such a directory in my mac, I replace the scripts with 'xattr -cr .' It works. However, I guess I need to add this script manually whenever I create a new project?

I delete this script, and the error pop up again after 10 mins.

**3. I tried add '.tmp' extension on openFramework folder (like this : of_v0.11.1_osx_release.tmp), which will stop syncing the specific folder to icloud.
[[https://appletoolbox.com/how-to-stop-icloud-syncing-desktop-folders/]](https://appletoolbox.com/how-to-stop-icloud-syncing-desktop-folders/).**
It works, and the error hasn't popped up again, at least two hours........

Sorry for the descriptions, I am not from a computer science background, so I don't know what happened behind the screen. I guess this problem should be related to the icloud syncing, and 'xattr -cr .' would like to fix the problem. I will be back if the error pops up again.

Many thanks!
